agent/recognition/binarymatch.py

```
@@ -281,3 +313,501 @@ def _try_recognition(self, context: Context, node_name: str,
+# 4. 对每个候选 blob，提取其包围框内的"内部亮色像素"（非红且亮）。
+# 5. 对内部亮色区域做垂直投影，检测感叹号特有的"竖线-断层-圆点"双段结构。
+# 6. 通过所有检查的第一个 blob 即命中，返回其在全图坐标系中的包围框。
+#
+# [为何此方案稳健]
+# - 红色区域连片不影响：感叹号白色仍被红色包围，拓扑关系不变。
+# - 任意背景：仅分析红色 blob 内部，外部背景完全隔离。
+# - 尺寸变化：无模板匹配，面积阈值范围宽松即可适配多种大小。
+# - 高光点误识别：高光点无垂直断层，双段检测可过滤。
+#
+# [参数说明]
+#   hsv_ranges   (list)  红色 HSV 范围，默认覆盖标准红色双段（H 绕 0 点）。
```

Ruff (0.15.14)

[warning] Comment contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF003)